zc_common/jwt_auth/permissions.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework.permissions import BasePermission
+from .utils import USER_ROLE, STAFF_ROLE, SERVICE_ROLE, ANONYMOUS_ROLE
+
+
+class IsUser(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        return request.user and request.user.is_authenticated() and USER_ROLE in user.roles
```

Comment:
You have `user` as a shortcut to `request.user`, and yet you didn't use it for all instances of `request.user`?